[demo/cartservice] Add support for logs (#3213)
open-telemetry/opentelemetry.io: content/en/docs/demo/services/cart.md

[Cart service source](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/cartservice/)

> **Note** OpenTelemetry for .NET uses the `System.Diagnostic.DiagnosticSource`

> > library as its API in lieu of the standard OpenTelemetry API for Traces and

Reviewer: ```suggestion
> library as its API instead of the standard OpenTelemetry API for Traces and
```

## Logs

> Logs should be configured in .NET dependency injection container on

Reviewer 2: ```suggestion
Logs are configured in the .NET dependency injection container on
```

> Logs should be configured in .NET DI container on `LoggingBuilder` level by

Reviewer 3: ```suggestion
Logs should be configured in .NET dependency injection container on `LoggingBuilder` level by
```

Author: d8b4adaa1ac036e66ee6057895dc657c2bb66368
